lib/vmpooler/providers/vsphere.rb

```diff
@@ -784,60 +784,24 @@ def find_snapshot(vm, snapshotname)
           get_snapshot_list(vm.snapshot.rootSnapshotList, snapshotname) if vm.snapshot
         end
 
-        def find_vm(vmname, connection)
-          find_vm_light(vmname, connection) || find_vm_heavy(vmname, connection)[vmname]
-        end
-
-        def find_vm_light(vmname, connection)
-          connection.searchIndex.FindByDnsName(vmSearch: true, dnsName: vmname)
+        def build_propSpecs(datacenter, folder, vmname)
+          propSpecs = {
+            entity => self,
+            :inventoryPath => "#{datacenter}/vm/#{folder}/#{vmname}"
+          }
+          propSpecs
         end
 
-        def find_vm_heavy(vmname, connection)
-          vmname = vmname.is_a?(Array) ? vmname : [vmname]
-          container_view = get_base_vm_container_from(connection)
-          property_collector = connection.propertyCollector
-
-          object_set = [{
-            obj: container_view,
-            skip: true,
-            selectSet: [RbVmomi::VIM::TraversalSpec.new(
-              name: 'gettingTheVMs',
-              path: 'view',
-              skip: false,
-              type: 'ContainerView'
-            )]
-          }]
-
-          prop_set = [{
-            pathSet: ['name'],
-            type: 'VirtualMachine'
-          }]
-
-          results = property_collector.RetrievePropertiesEx(
-            specSet: [{
-              objectSet: object_set,
-              propSet: prop_set
-            }],
-            options: { maxObjects: nil }
-          )
-
-          vms = {}
-          results.objects.each do |result|
-            name = result.propSet.first.val
-            next unless vmname.include? name
-            vms[name] = result.obj
-          end
-
-          while results.token
-            results = property_collector.ContinueRetrievePropertiesEx(token: results.token)
-            results.objects.each do |result|
-              name = result.propSet.first.val
-              next unless vmname.include? name
-              vms[name] = result.obj
-            end
-          end
+        def find_vm(pool_name, vmname, connection)
+          pool_configuration = pool_config(pool_name)
```

Comment:
this method can return nil so we should protect against that and maybe log an error message if it happens

Reply:
I think it's more about pool_configuration being `nil` and the next line `pool_configuration['folder']` will raise a method error.

* Should either raise an error it pool_configuration is nil
`raise("could not find pool_name") if pool_configuration.nil?`

or

* Return a nil value
`return nil if pool_configuration.nil?`

Reply:
Should probably document the expected return values in code comments for future-us

Reply:
Actually given;

`expect(subject.find_vm(poolname,missing_vm,connection)).to be_nil` in the spec tests, it should return `nil`

Reply:
Ah ok, I follow now. I'll update this and push a change.